apps/aecontract/src/aect_state_tree.erl

```diff
@@ -51,24 +52,74 @@ insert_contract(Contract, Tree = #contract_tree{ contracts = CtTree }) ->
     Id         = aect_contracts:id(Contract),
     Serialized = aect_contracts:serialize(Contract),
     CtTree1    = aeu_mtrees:insert(Id, Serialized, CtTree),
-    Tree#contract_tree{ contracts = CtTree1 }.
+    CtTree2    = insert_store(Contract, CtTree1),
+    Tree#contract_tree{ contracts = CtTree2 }.
+
+insert_store(Contract, CtTree) ->
+    Id = aect_contracts:store_id(Contract),
+    Store = aect_contracts:state(Contract),
+    insert_store_nodes(Id, Store, CtTree).
+
+insert_store_nodes(Prefix, Store, CtTree) ->
```

Comment:
If I remember correctly, writing `0` to the store, will end up deleting the store position. If this is is implemented by removing the store position in the map, this insertion into the MPT will preserve the deleted value (as the previous value).

This needs to be special treated if the semantics is to be preserved. One way of doing this would be to keep the position in the map and use the `<<>>` value, as this will delete the value in the MPT. This only works if the value `<<>>` isn't a valid value for the store, but if it is a valid value, this in turn needs to be special treated.

Reply:
Exactly, but since the tree node of the contract will still point to the old store in the MPT, we need to explicitly delete these positions (or equivalently overwrite them with `<<>>`). Otherwise they will remain in the tree. The subtree to the contract is not fresh in every run.

Reply:
Fixed in enter_store...